[SDK-1761] Updated Podfile to use latest core SDK

nishant-clevertap opened the pull request:
- Updated cocoapods examples to point to latest core SDK.
- Updated Readme and Changelog.

nishant-clevertap committed:
chore(SDK-1761): Updated podfile to use latest core SDK

- Updated cocoapods examples to point to latest core SDK.
- Updated Readme and Changelog.

akashvercetti commented on CHANGELOG.md:
```diff
@@ -1,6 +1,9 @@
 All notable changes to this project will be documented in this file.
 
+### [Version 1.0.2](https://github.com/CleverTap/clevertap-geofence-ios/releases/tag/1.0.2) (July 12, 2022)
+* Compatible with latest CleverTap iOS SDK.
```
please mention the latest iOS SDK version v4.1.1

nishant-clevertap committed:
chore(SDK-1761): Updated pod to use CleverTap-Geofence-SDK

- Updated pod name to CleverTap-Geofence-SDK
- Added module name for pod to generate same module name - CleverTapGeofence.
- Updated example Podfiles to use new pod.
- Updated Readme.

nishant-clevertap committed:
chore(SDK-1761): Updated version in podspec

- Updated version number in podspec file.
- Updated Changelog.

darshanclevertap reviewed (approved):
👍🏾 LGTM
